docs/cpp/configure-intellisense.md

```diff
@@ -0,0 +1,192 @@
+<div class="details collapse">
+<div class="comment"><p> Using the default install path for GCC:
+```json
+{
+  "configurations": [
+        {
+          "name": "Linux-GCC",
+          "includePath": [
+            "${workspaceFolder}/**"
+          ],
+          "defines": [],
+          "compilerPath": "/usr/bin/g++",
+          "cStandard": "c17",
+          "cppStandard": "c++17",
+          "intelliSenseMode": "gcc-x64",
+          "browse": {
+            "path": [
+              "${workspaceFolder}"
+            ],
+            "limitSymbolsToIncludedHeaders": true,
+            "databaseFilename": ""
+          }
+        }
+  ],
+  "version": 4
+}
+```
+</p>
+</div>
+</div>
+
+## Project Level IntelliSense Configuration
+
+Configuring IntelliSense with a compiler provides you with core IntelliSense features. This steup is called the base configuration. For more complex usage scenarios, such as setting up a project that requires:
+
+- additional include paths, such as references to one or multiple different libraries
+- specific compiler arguments that influence the behavior of the language(and therefore IntelliSense)
+
+There are multiple other ways to configure IntelliSense. You can provide these additional configurations either through the `c_cpp_properties.json` file and related settings, a custom configuration provider in the form of another VS Code extension (for example, the [Makefile Tools](https://marketplace.visualstudio.com/items?itemName=ms-vscode.makefile-tools) or [CMake Tools](/docs/cpp/CMake-linux.md) extension), or a `compile_commands.json` file.
+
+### Configuration Providers
+
+A custom configuration provider is another extension in VS Code which can potentially provide more accurate C++ IntelliSense configuration than the C/C++ Extension. For example, for the CMake or Make build systems, where the [Makefile Tools](https://marketplace.visualstudio.com/items?itemName=ms-vscode.makefile-tools) or [CMake Tools](/docs/cpp/CMake-linux.md) extensions can be configuration providers. To add an extension as a configuration provider, either select the extension through the configuration quick pick, add it to configuration UI by editing the **Configuration provider** field under **Advanced Settings**, or add the `configurationProvider` field to your `c_cpp_properties.json` file. For example, for the CMake extension, the path to add would be `ms-vscode.cmake-tools`.
+
+The C/C++ Extension scans your system for custom configuration providers. If it identifies only one custom configuration provider, this configuration provider is automatically configured for IntelliSense. If there are multiple configuration providers identified, you need to select which the Extension should use by opening [the configuration quick pick](/docs/cpp/configure-intellisense.md#_Option-1.-Select-a-configuration-option-through-the-configuration-quick-pick).
+
+### `compile_commands.json` File
```

Comment:
inline code blocks don't render well in H2 headers so "compile_command.json file"